pkg/readers/container_image_reader.go

```diff
@@ -0,0 +1,148 @@
+package readers
+
+import (
+	"context"
+	"fmt"
+	scalibr "github.com/google/osv-scalibr"
+	scalibrlayerimage "github.com/google/osv-scalibr/artifact/image/layerscanning/image"
+	"github.com/google/osv-scalibr/converter"
+	"github.com/google/osv-scalibr/extractor"
+
+	//"github.com/google/osv-scalibr/converter"
+	el "github.com/google/osv-scalibr/extractor/filesystem/list"
+	sl "github.com/google/osv-scalibr/extractor/standalone/list"
+	"github.com/safedep/vet/pkg/common/logger"
+	"github.com/safedep/vet/pkg/models"
+	"github.com/safedep/vet/pkg/parser"
+)
+
+type ContainerImageReaderConfig struct {
+	Image string
+}
+
+type containerImageReader struct {
+	config *ContainerImageReaderConfig
+}
+
+var _ PackageManifestReader = &containerImageReader{}
+
+func NewContainerImageReader(config *ContainerImageReaderConfig) (*containerImageReader, error) {
```

Comment:
We should separate the config and the image target. They are different. `ImageTarget` such as `node:latest` is the target of scan / reading while config defines "how" it is read. I suggest take 2 param as input - `config` and `imageRef`. Fail fast if the `imageRef` is invalid in `NewContainerImageReader`